File: script_perf/factorial/lua/factorial.lua
#!/usr/bin/lua

function recur_fact(n)
	if n == 0 then
		return 1
	else 
		return n*recur_fact(n-1)
	end
end

function factorial(n)
	if n == 0 then
		return 1
	else
		ret = 1
		for i = 1, n do
			ret = ret * i
		end
		return ret
	end
end

if not arg[1] then
	print("Please specify the number which need to calculat its factorial")
	return
end

loops = tonumber(arg[2])
n = tonumber(arg[1])

if not loops then 
	loops = 1
end


for i = 1, loops do
	ret = factorial(n)
end

File: script_perf/factorial/cpp/factorial.cc
/*
 * =====================================================================================
 *
 *       Filename:  factorial.cc
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  04/17/2016 09:35:37 PM
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  YOUR NAME (), 
 *        Company:  
 *
 * =====================================================================================
 */
#include <cstdlib>
#include <iostream>

using namespace std;

typedef unsigned long long uint64_t;

uint64_t factorial(int n)
{
	if (n == 0) {
		return 1;
	}

	uint64_t ret = 1;
	for (int i = 1; i <= n; ++i) {
		ret = ret*i;
	}

	return ret;
}

int main(int argc, char **argv)
{
	if (argc < 2) {
		cerr << "Please specify the number which need calculate its factorial" << endl;
		return 1;
	}

	int num = atoi(argv[1]);
	int loops = 1;
	if (argc >= 3) {
		loops = atoi(argv[2]);
	}

	uint64_t ret;
	for (int i = 0; i < loops; ++i) {
		ret = factorial(num);
	}

	cout << ret << endl;
	return 0;
}
